spring-rabbit/src/main/java/org/springframework/amqp/rabbit/AsyncRabbitTemplate.java

```diff
@@ -713,8 +722,10 @@ void setNackCause(String nackCause) {
 
 		void startTimer() {
 			if (AsyncRabbitTemplate.this.receiveTimeout > 0) {
-				this.timeoutTask = AsyncRabbitTemplate.this.taskScheduler.schedule(new TimeoutTask(), // NOSONAR sync
-						new Date(System.currentTimeMillis() + AsyncRabbitTemplate.this.receiveTimeout));
+				synchronized (AsyncRabbitTemplate.this.taskScheduler) {
```

Comment:
That was a mistake - should be `synchronized (ART.this)` - to match the other synchronizations. Will push in the morning (or you can fix during merge if this is the only problem).